app/src/components/market/market_routes.tsx

```diff
@@ -54,30 +54,20 @@ const MarketValidateContractAddress: React.FC<Props> = (props: Props) => {
   return (
     <Switch>
       <Route exact path="/:address" component={MarketDetailsPage} />
-      <Route exact path="/:address/buy" component={MarketBuyPage} />
-      <Route exact path="/:address/sell" component={MarketSellPage} />
-      <Route exact path="/:address/fund" component={MarketFundPage} />
+      {!account ? (
+        <MessageNeedConnection />
+      ) : (
```

Comment:
I have my concerns about it, if you have any better way to do this, suggestions welcome